skin/src/vue/c-content.vue

```diff
@@ -83,12 +89,18 @@ const title = computed(() => {
 const cristal: CristalApp = inject<CristalApp>("cristal")!;
 
 async function fetchPage(page: string) {
+  if (page != null) {
+    page = encodeURIComponent(page);
+  }
   loading.value = true;
   try {
     // Provides a reactive variable to be updated if the page content is updated
     // in the background.
     cristal.setContentRef(currentPage);
     currentPage.value = await cristal.getPage(page || currentPageName.value);
+    breadcrumbItems.value = await cristal
+      .getWikiConfig()
+      .pageHierarchyResolver.getPageHierarchy(currentPage.value!);
```

Comment:
I suggest introducing a `PageHierarchyResolverProvider` component that would only have a get() method returning the correct `PageHierarchyResolver` components based on the current wiki config (and the callback).
The code in the vue component would just be:

```suggestion
    breadcrumbItems.value = await cristal
      .getContainer().get<PageHierarchyResolverProvider>("PageHierarchyResolverProvider")
      .get().getPageHierarchy(currentPage.value!);
```